extensions/ruby/src/language_servers/solargraph.rs

```diff
@@ -0,0 +1,121 @@
+use zed::lsp::{Completion, CompletionKind, Symbol, SymbolKind};
+use zed::{CodeLabel, CodeLabelSpan};
+use zed_extension_api::{self as zed, Result};
+
+pub struct Solargraph {}
+
+impl Solargraph {
+    pub const LANGUAGE_SERVER_ID: &'static str = "solargraph";
+
+    pub fn new() -> Self {
+        Self {}
+    }
+
+    pub fn server_script_path(&mut self, worktree: &zed::Worktree) -> Result<String> {
+        let path = worktree
+            .which("solargraph")
+            .ok_or_else(|| "solargraph must be installed manually".to_string())?;
+
+        Ok(path)
+    }
+
+    pub fn label_for_completion(&self, completion: Completion) -> Option<CodeLabel> {
+        match completion.kind? {
+            CompletionKind::Method => {
+                let highlight_name = match completion.kind? {
+                    CompletionKind::Class | CompletionKind::Module => "type",
+                    CompletionKind::Constant => "constant",
+                    CompletionKind::Method => "function.method",
+                    CompletionKind::Keyword => {
+                        if completion.label.starts_with(":") {
+                            "string.special.symbol"
+                        } else {
+                            "keyword"
+                        }
+                    }
+                    CompletionKind::Variable => {
+                        if completion.label.starts_with("@") {
+                            "property"
+                        } else {
+                            return None;
+                        }
+                    }
+                    _ => return None,
+                };
+
+                let len = completion.label.len();
+                let name_span =
+                    CodeLabelSpan::literal(completion.label, Some(highlight_name.to_string()));
+
+                Some(CodeLabel {
+                    code: Default::default(),
+                    spans: if let Some(detail) = completion.detail {
```

Comment:
Added signature output to the completion:

![CleanShot 2024-05-04 at 09 17 37@2x](https://github.com/zed-industries/zed/assets/1894248/0f5696a2-5b0e-497a-b5e9-923c7974baf0)